### PR TITLE
CI-Reporter testgrid logic into pkg/testgrid

### PR DESCRIPTION
**Pull request by leonardpahlke**
#### What type of PR is this?

/kind cleanup
/kind feature

#### What this PR does / why we need it:

This PR continuous to clean up the ci-signal reporter tool as discussed in #2316 (align testgrid pkg with testgrid reporter methods (align with k/release) + add tests).

* Move testgrid logic from cmd/ci-reporter/testgrid to pkg/testgrid/testgrid-scraper
* Add tests pkg/testgrid/testgrid-scraper_test
* Add concurrency if multiple dashboards are being requested
* Extended testgrid report if the short flag is not set (see user-facing changes)

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:

Open clean-up task after this one
* align github pkg with github reporter methods (align with k/release) + add tests

The new logic in testgrid/testgrid-scraper does not leverage from the existing testgrid/testgrid logic. 
The testgrid.go logic works with a testgrid config which is stored here: https://storage.googleapis.com/k8s-testgrid/config. I dont have access to the config. I think it would be nice if everybody could use the ci-signal tool without any process needed to given them access rights. JSON parsing from endpoints like https://testgrid.k8s.io/sig-release-master-informing/summary does not require authorization (as the website is public). I am curios what everybody things about that. 

#### Does this PR introduce a user-facing change?

```release-note
Extended testgrid report from the ci-reporter if the short flag is not set that includeds details about failing and flaky tests
```

/label tide/merge-method-squash
/assign @puerco

### CODE REVIEW

File: cmd/ci-reporter/cmd/testgrid.go [hunk 1]
```diff
@@ -17,97 +17,84 @@ limitations under the License.
...
+				// Filter sigs
+				sigRegex := regexp.MustCompile(`sig-[a-zA-Z]+`)
+				sigsInvolved := map[string]int{}
+				for i := range jobData.Tests {
+					sigs := sigRegex.FindAllString(jobData.Tests[i].TestName, -1)
+					for _, sig := range sigs {
+						sigsInvolved[sig]++
+					}
+				}
+				sigs := reflect.ValueOf(sigsInvolved).MapKeys()
```
Comment (puerco) [inline cmd/ci-reporter/cmd/testgrid.go:80]: Let's iterate this one and save the whole reflect import:
```suggestion
				sigs := []string{}
				for k := range sigsInvolved {
					sigs = append(sigs, k)
				}
```

File: cmd/ci-reporter/cmd/root.go [hunk 2]
```diff
@@ -89,7 +89,10 @@ func RunReport(cfg ReporterConfig) error {
 	if err != nil {
 		return err
 	}
-	PrintTestgridReportData(testgridReportData)
+	err = PrintTestgridReportData(cfg, &testgridReportData)
+	if err != nil {
+		return err
+	}
```
Comment (puerco) [inline cmd/ci-reporter/cmd/root.go:95]: nit
```suggestion
	if err = PrintTestgridReportData(cfg, &testgridReportData); err != nil {
		return err
	}
```

File: cmd/ci-reporter/cmd/testgrid.go [hunk 3]
```diff
@@ -17,97 +17,84 @@ limitations under the License.
 package cmd
 
 import (
-	"encoding/json"
 	"fmt"
-	"io/ioutil"
-	"net/http"
-)
+	"reflect"
```
Comment (puerco) [inline cmd/ci-reporter/cmd/testgrid.go:21]: (see reflection below)
```suggestion
```

**Review by puerco — CHANGES_REQUESTED** [review]
Thanks @leonardpahlke ! Just some tiny details but looks good :point_down:

File: cmd/ci-reporter/cmd/testgrid.go [hunk 4]
```diff
@@ -17,97 +17,84 @@ limitations under the License.
...
-		if err != nil {
-			return nil, err
-		}
+	return testgrid.ReqTestgridDashboardSummaries(testgridURLs)
+}
 
-		jobs := make(map[string]overview)
-		err = json.Unmarshal(body, &jobs)
+// PrintTestgridReportData used to print testgrid report data out to the console
+func PrintTestgridReportData(cfg ReporterConfig, stats *testgrid.DashboardData) error {
```
[the remark this answers is not in the document]
Comment (leonardpahlke) [reply cmd/ci-reporter/cmd/testgrid.go:43]: Hm, yes! Maybe we could combine that with revisiting [tablewriter](https://github.com/olekukonko/tablewriter). I can put this on project board.

File: cmd/ci-reporter/cmd/root.go (hunk 2 above)
Comment (leonardpahlke) [reply cmd/ci-reporter/cmd/root.go:95]: I am getting this linting warning:
sloppyReassign: re-assignment to `err` can be replaced with `err := PrintTestgridReportData(cfg, &testgridReportData)`

File: cmd/ci-reporter/cmd/testgrid.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (leonardpahlke) [reply cmd/ci-reporter/cmd/testgrid.go:42]: I created two notes:
Generic template: https://github.com/orgs/kubernetes/projects/65#card-74561975
Tablewriter printer: https://github.com/orgs/kubernetes/projects/65#card-74562132

FYI: I will turn the notes into issues after I done the refactoring of the github report (move current github report logic into the related github ppackages)

File: cmd/ci-reporter/cmd/root.go (hunk 2 above)
Comment (puerco) [reply cmd/ci-reporter/cmd/root.go:95]: oops yes my bad there

File: cmd/ci-reporter/cmd/testgrid.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (puerco) [reply cmd/ci-reporter/cmd/testgrid.go:42]: Sounds good, thanks :+1:

**Comment by puerco** [comment]
Thanks! @leonardpahlke!
/lgtm
/approve

**Pull request merged**